lightning/src/chain/channelmonitor.rs

```diff
@@ -166,12 +166,14 @@ pub struct HTLCUpdate {
 	pub(crate) payment_hash: PaymentHash,
 	pub(crate) payment_preimage: Option<PaymentPreimage>,
 	pub(crate) source: HTLCSource,
+	pub(crate) sink_channel_id: Option<[u8; 32]>,
 	pub(crate) onchain_value_satoshis: Option<u64>,
```

Comment:
@TheBlueMatt i've added `sink_channel_id but` do you think it's worth moving `source_channel_id` here as well? Without it the implementation works but feels half-baked. Similarly, could work on adding source/sink node ids in this PR.

Reply:
Instead, lets have `chain::Watch` return the `OutPoint` for the channel the event came from.

Reply:
I could return a `Vec<(OutPoint, MonitorEvent)>` from `release_pending_monitor_events()` to associate an `OutPoint` with an event. That should remove the `OutPoint` dependency from the other `MonitorEvent` enum fields too. Is this what you are suggesting?

Reply:
I think the problem with that is it'll ignore the already present pending events, no? So what I could do is sort that at source, append the `monitor.get_and_clear_pending_monitor_events()` vec to the corresponding `OutPoint` tuple and then return it.

Also, it seems like order doesn't matter so we could use `HashMap<OutPoint, Vec<MonitorEvent>>`. `OutPoint` keys will be unique and we get an io speed bump.

Reply:
Small thing but thought I should let you know, ignoring already present pending events was causing tests to fail. Sadly, have to clone (which is done using `split_off(0)` [here](https://github.com/lightningdevkit/rust-lightning/pull/1475/files#diff-dfdfb24521752bd4735cf117dc1047702c5057f88b1a8526de1c8fb785c1d18dR672), not sure why).